src/librustc_typeck/check/op.rs

```diff
@@ -167,14 +155,16 @@ fn check_overloaded_binop<'a, 'tcx>(fcx: &FnCtxt<'a, 'tcx>,
                                     lhs_expr: &'tcx hir::Expr,
                                     lhs_ty: Ty<'tcx>,
                                     rhs_expr: &'tcx hir::Expr,
-                                    op: hir::BinOp)
+                                    op: hir::BinOp,
+                                    assign: bool)
```

Comment:
Nit: can we newtype this bool? I think I prefer something like:

```
struct IsAssign(bool)
```

and then in the call `foo(..., IsAssign(true))`, though `IsAssign::Yes` is also ok.